File: Lv0/자릿수더하기/Solution.java
package Lv0.자릿수더하기;

class Solution {
    public int solution(int n) {
        int answer = 0;

        // int n을 문자열로 변환 -> 배열로 만들기 위해서
        String s = String.valueOf(n);

        // 각 자리 숫자를 더함
        for(char digit : s.toCharArray()) {
            answer += Character.getNumericValue(digit);
        }

        return answer;
    }
}

/* toCharArray()
*   Java에서 문자열 객체를 문자형 배열로 변환하는데 사용한다 반환타입은 char[] 이다.
*
* Character.getNumericValue(digit)
*   주머진 문자를 숫자로 변환한다.
* */

File: Lv0/자릿수더하기/Solution.js
function solution(n) {
    let answer = 0;
    answer = String(n) //n을 문자열로 변환
        .split('') // 각 문자로 나누어 배열전환
        .map(Number) // 각 문자를 숫자로 변환
        .reduce((a,b) => a+b,0) //배열의 숫자들을 합산.
    return answer;
}